src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -178,12 +181,22 @@ class Container extends React.Component<Props, ComponentState> {
                     title: nextProps.intl.formatMessage({
                         id: FM.ENTITYCREATOREDITOR_TITLE_EDIT,
                         defaultMessage: 'Edit Entity'
-                    })
+                    }),
+                    enumValues: this.initEnumValues(nextProps.entity.enumValues)
                 })
             }
         }
     }
 
+    initEnumValues(enumValues: CLM.EnumValue[] | undefined): (CLM.EnumValue | null)[] {
+        if (!enumValues) {
+            return Array(CLM.MAX_ENUM_VALUES).fill(null)
+        }
+        const enumClone: CLM.EnumValue[] = JSON.parse(JSON.stringify(enumValues))
```

Comment:
Why is the clone required?

Is this so they can edit the values in the view without updating the actual original values?

We should try to limit the use of `JSON.parse(JSON.stringify(`.  It's a rather expensive hack to clone things in javascript so we want to only use it when needed.

Reply:
Yeah - need a deep copy.  Spread just copies the object pointers.